weed/server/master_server.go

```diff
@@ -384,17 +404,18 @@ func (ms *MasterServer) OnPeerUpdate(update *master_pb.ClusterNodeUpdate, startF
 					})
 					if err != nil {
 						glog.Warningf("failed to removing old raft server %s: %v", peerName, err)
+						return
 					}
 					glog.V(0).Infof("old raft server %s removed", peerName)
 					return
 				case peerDone := <-ms.onPeerUpdateDoneCn:
```

Comment:
this may not work as expected. The channel is not broadcasting. Only one of the goroutine will get an peerDone here, but it may not be the goroutine that is waiting for this.